### ISSUE TITLE
Installating OKD 4.8.0-0.okd-2021-10-01-221835 failed on UPI Openstack

### ISSUE THREAD

**Issue by m-yosefpor** [openshift/okd#906]
**Describe the bug**

When installing 4.8.0-0.okd-2021-10-01-221835 on UPI Openstack, all Cluster Operators stuck in CrashLoopBackOff state with the following in the logs:

```
unable to load configmap based request-header-client-ca-file: Get "https://172.30.0.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authenticat
ion": dial tcp 172.30.0.1:443: i/o timeout
```
 
 
bootkube.serive logs:
```
Oct 03 23:31:48 host-172-21-51-127 bootkube.sh[17571]: Waiting up to 20m0s for the Kubernetes API
Oct 03 23:31:49 host-172-21-51-127 bootkube.sh[17571]: API is up
Oct 03 23:31:49 host-172-21-51-127 bootkube.sh[17571]: Skipped "0000_00_cluster-version-operator_00_namespace.yaml" namespaces.v1./openshift-cluster-version -n  as it already exists
.... (many skipped lines)
Oct 03 23:32:09 host-172-21-51-127 bootkube.sh[17571]: Skipped "secret-service-network-serving-signer.yaml" secrets.v1./service-network-serving-signer -n openshift-kube-apiserver-operator as it already exists
```

Nodes are ready:

```
$ oc get no
NAME             STATUS   ROLES    AGE   VERSION
okd4-masters-0   Ready    master   39m   v1.22.1+8719299-1678
okd4-masters-1   Ready    master   39m   v1.22.1+8719299-1678
okd4-masters-2   Ready    master   40m   v1.22.1+8719299-1678
```

Network pods are running:

```
$ oc get po -A | grep Run
openshift-machine-config-operator                  machine-config-operator-9d768655f-kb5k7                   1/1     Running             0          45m
openshift-multus                                   multus-additional-cni-plugins-nx244                       1/1     Running             0          39m
openshift-multus                                   multus-additional-cni-plugins-qvxng                       1/1     Running             0          39m
openshift-multus                                   multus-additional-cni-plugins-qvzds                       1/1     Running             0          39m
openshift-multus                                   multus-mgfzv                                              1/1     Running             0          39m
openshift-multus                                   multus-rhjb9                                              1/1     Running             0          39m
openshift-multus                                   multus-rstpx                                              1/1     Running             0          39m
openshift-network-diagnostics                      network-check-target-7cj27                                1/1     Running             0          39m
openshift-network-diagnostics                      network-check-target-dcpft                                1/1     Running             0          39m
openshift-network-diagnostics                      network-check-target-kvrmb                                1/1     Running             0          39m
openshift-network-operator                         network-operator-64bb648c5b-hhgts                         1/1     Running             2          46m
openshift-ovn-kubernetes                           ovnkube-master-l4qnl                                      6/6     Running             5          39m
openshift-ovn-kubernetes                           ovnkube-master-rlkn2                                      6/6     Running             5          39m
openshift-ovn-kubernetes                           ovnkube-master-z2s7l                                      6/6     Running             10         39m
openshift-ovn-kubernetes                           ovnkube-node-6mv8d                                        4/4     Running             0          39m
openshift-ovn-kubernetes                           ovnkube-node-m6qkh                                        4/4     Running             0          39m
openshift-ovn-kubernetes                           ovnkube-node-mwdrb                                        4/4     Running             1          39m
```

Almost none of COs are ready:

```
$ oc get co
NAME                                       VERSION                         AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication
baremetal
cloud-controller-manager
cloud-credential                                                           True        False         False      46m
cluster-autoscaler
config-operator
console
csi-snapshot-controller
dns
etcd
image-registry
ingress
insights
kube-apiserver
kube-controller-manager
kube-scheduler
kube-storage-version-migrator
machine-api
machine-approver
machine-config
marketplace
monitoring
network                                    4.8.0-0.okd-2021-10-01-221835   True        True          False      32m
node-tuning
openshift-apiserver
openshift-controller-manager
openshift-samples
operator-lifecycle-manager
operator-lifecycle-manager-catalog
operator-lifecycle-manager-packageserver
service-ca
storage
```

Because all cluster operator pod are in CrashLoopBackOff state:

```
NAMESPACE                                         NAME                                                     READY  STATUS             RESTARTS  AGE  NODE
openshift-apiserver-operator                      openshift-apiserver-operator-5496f56bcd-fqtb5            0/1    CrashLoopBackOff   8         43m  okd4-masters-0
openshift-authentication-operator                 authentication-operator-699c564dbd-9sz9k                 0/1    CrashLoopBackOff   11        43m  okd4-masters-0
openshift-cloud-controller-manager-operator       cluster-cloud-controller-manager-74cd748f5-wg7n8         0/1    CrashLoopBackOff   9         43m  okd4-masters-0
openshift-cloud-credential-operator               cloud-credential-operator-59468b75bc-b877g               0/2    ContainerCreating  0         43m  okd4-masters-1
openshift-cluster-machine-approver                machine-approver-5659df9c4f-tk924                        0/2    ContainerCreating  0         43m  okd4-masters-1
openshift-cluster-node-tuning-operator            cluster-node-tuning-operator-7cc784b8c4-b22s9            0/1    ContainerCreating  0         43m  okd4-masters-0
openshift-cluster-storage-operator                cluster-storage-operator-86c75b74d5-75vcm                0/1    CrashLoopBackOff   8         43m  okd4-masters-0
openshift-cluster-storage-operator                csi-snapshot-controller-operator-95d46c8bd-vjx4k         0/1    CrashLoopBackOff   8         43m  okd4-masters-1
openshift-cluster-version                         cluster-version-operator-6c85f4d56f-dzlr6                0/1    ContainerCreating  0         44m  okd4-masters-2
openshift-config-operator                         openshift-config-operator-64796888c7-52ltf               0/1    CrashLoopBackOff   13        43m  okd4-masters-1
openshift-controller-manager-operator             openshift-controller-manager-operator-689946898f-zpmq2   0/1    CrashLoopBackOff   8         43m  okd4-masters-1
openshift-dns-operator                            dns-operator-5bb5f7f68-s7zkk                             0/2    ContainerCreating  0         43m  okd4-masters-2
openshift-etcd-operator                           etcd-operator-7689b47698-x9jjh                           0/1    CrashLoopBackOff   8         43m  okd4-masters-2
openshift-image-registry                          cluster-image-registry-operator-7ffdb7f654-5qvbq         0/1    ContainerCreating  0         43m  okd4-masters-2
openshift-ingress-operator                        ingress-operator-66755d8bf6-xmsqf                        0/2    ContainerCreating  0         43m  okd4-masters-0
openshift-insights                                insights-operator-54cbcd8d49-td29l                       0/1    CrashLoopBackOff   8         43m  okd4-masters-0
openshift-kube-apiserver-operator                 kube-apiserver-operator-86d9f579d7-4d9z8                 0/1    CrashLoopBackOff   8         43m  okd4-masters-1
openshift-kube-controller-manager-operator        kube-controller-manager-operator-6c77db5957-pv7v2        0/1    CrashLoopBackOff   8         43m  okd4-masters-2
openshift-kube-scheduler-operator                 openshift-kube-scheduler-operator-7c7b8c4f69-wkxc6       0/1    CrashLoopBackOff   8         43m  okd4-masters-2
openshift-kube-storage-version-migrator-operator  kube-storage-version-migrator-operator-5f784db8d9-zwdql  0/1    CrashLoopBackOff   8         43m  okd4-masters-1
openshift-machine-api                             cluster-autoscaler-operator-cd49cddb9-95bnj              0/2    ContainerCreating  0         43m  okd4-masters-1
openshift-machine-api                             cluster-baremetal-operator-64dfbb9f68-xz9xt              0/2    ContainerCreating  0         43m  okd4-masters-2
openshift-machine-api                             machine-api-operator-6b9b57586d-qtvg2                    0/2    ContainerCreating  0         43m  okd4-masters-0
openshift-marketplace                             marketplace-operator-6d7bf9b949-xqdgx                    0/1    ContainerCreating  0         43m  okd4-masters-1
openshift-monitoring                              cluster-monitoring-operator-dfcb4f59f-m89qj              0/2    CrashLoopBackOff   19        43m  okd4-masters-2
```

All of them with the same log:
```
unable to load configmap based request-header-client-ca-file: Get "https://172.30.0.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authenticat
ion": dial tcp 172.30.0.1:443: i/o timeout
```

Configmap exists:

```
$ oc get -n kube-system cm extension-apiserver-authentication
NAME                                 DATA   AGE
extension-apiserver-authentication   6      68m
```

But api server not accessible with clusterIP from masters or bootstrap node:
```
 $ curl https://172.30.0.1:443 -k
 // timeout
 ```

**Version**

4.8.0-0.okd-2021-10-01-221835
UPI Openstack
OS: fedora-coreos-34.20210626.3.1-openstack.x86_64.qcow2
(OS downloaded from `./openshift-install coreos print-stream-json | jq -r '.architectures.x86_64.artifacts.openstack.formats."qcow2.xz".disk.location'` )

**How reproducible**

100%

Boostrapping cluster fails with new released OKD version 4.8.0-0.okd-2021-10-01-221835.

**Log bundle**

` $ oc adm must-gather`

[must-gather.local.8277507614122635598.tar.gz](https://github.com/openshift/okd/files/7274698/must-gather.local.8277507614122635598.tar.gz)

```
$  openshift-install gather bootstrap --dir=.    --bootstrap <bootstrap_address> \ 
    --master <master_1_address> \ 
    --master <master_2_address> \ 
    --master <master_3_address>"
```
[log-bundle-20211003235230.tar.gz](https://github.com/openshift/okd/files/7274733/log-bundle-20211003235230.tar.gz)

**Comment by m-yosefpor** [comment]
Isn't it a circular dependency? kube-apiserver-operator needs to contact `172.30.0.1`  to load configmap based request-header-client-ca-file , but it's not possible since api server pods are not running, because kube-apiserver-operator needs to install static pods resources for them. Also The bootstrap api server can not be accessed via `172.30.0.1` since there are no OVN pods nor `openvswitch.service` on bootstrap node. Is my understanding correct?

**Comment by m-yosefpor** [comment]
> No control plane info is collected. Nodes look off: `172.21.43.128` is `okd4-masters-0` and log-bundle has `172.21.51.128` record

All nodes have two interfaces. 172.21.51.128 and 172.21.43.128 are both IPs of `ens3` and `ens4` of `okd4-masters-0`

**Comment by m-yosefpor** [comment]
OVN has much dependencies to default gateway and it is not straight forward to make the non default gateway as primary interface for underlay connection. Until now we have made it work with many hack, but as we do not have flexibility to edit OVN manifests as demand (since it is managed by CNO), it is hard to maintain the hacks across upgrades.

Because if this, and many other reasons (more importantly observability, advanced network policy, cluster mesh and performance), we decided to switch to Cilium. This issue was fixed with that change too.
